sql/parse/parse.go

```diff
@@ -1354,6 +1358,75 @@ func selectExprsToExpressions(ctx *sql.Context, se sqlparser.SelectExprs) ([]sql
 	return exprs, nil
 }
 
+// StringToColumnDefaultValue takes in a string representing a default value and returns the equivalent Expression.
+func StringToColumnDefaultValue(ctx *sql.Context, exprStr string) (*sql.ColumnDefaultValue, error) {
+	// all valid default expressions will parse correctly with SELECT prepended, as the parser will not parse raw expressions
+	stmt, err := sqlparser.Parse("SELECT " + exprStr)
+	if err != nil {
+		return nil, err
+	}
+	//TODO: change these from fmt.Errorf
+	parserSelect, ok := stmt.(*sqlparser.Select)
+	if !ok {
+		return nil, fmt.Errorf("DefaultStringToExpression expected sqlparser.Select but received %T", stmt)
+	}
+	if len(parserSelect.SelectExprs) != 1 {
+		return nil, fmt.Errorf("default string does not have only one expression")
+	}
+	aliasedExpr, ok := parserSelect.SelectExprs[0].(*sqlparser.AliasedExpr)
+	if !ok {
+		return nil, fmt.Errorf("DefaultStringToExpression expected *sqlparser.AliasedExpr but received %T", parserSelect.SelectExprs[0])
+	}
+	parsedExpr, err := exprToExpression(ctx, aliasedExpr.Expr)
+	if err != nil {
+		return nil, err
+	}
+	// The literal and expression distinction seems to be decided by the presence of parentheses, even for defaults like NOW() vs (NOW())
+	// 2+2 would evaluate to a literal under the parentheses check, but will have children due to being an Arithmetic expression, thus we check for children.
+	return ExpressionToColumnDefaultValue(ctx, parsedExpr, len(parsedExpr.Children()) == 0 && !strings.HasPrefix(exprStr, "("))
+}
+
+// ExpressionToColumnDefaultValue takes in an Expression and returns the equivalent ColumnDefaultValue if the expression
+// is valid for a default value. If the expression represents a literal (and not an expression that returns a literal, so "5"
+// rather than "(5)"), then the parameter "isLiteral" should be true.
+func ExpressionToColumnDefaultValue(ctx *sql.Context, inputExpr sql.Expression, isLiteral bool) (*sql.ColumnDefaultValue, error) {
+	expr, err := expression.TransformUp(inputExpr, func(e sql.Expression) (sql.Expression, error) {
```

Comment:
There needs to be a really, really good reason to do this here instead of at analyzer time. Expressions aren't expected to be eval-able at parse time, which is what we're doing here.

Reply:
I'll address all of the analyzer-related comments here. The reason that I am not doing this is that it ties any integrator that wants to make use of default values to the SQL write path, unless we create a separate entry point in `go-mysql-server` that's specifically for default values.

As an example, in the `dolt` project we have an import path for CSV files, and it currently does not make use of anything from the SQL path. With the way that we're planning on implementing default values (where the integrator stores a string), the only way for them to make use of the default value is with a reliance on the SQL engine. This implementation is geared so that the reliance is on a function call, rather than forcing all write paths through the SQL engine (which I don't find a reasonable solution).

We could make a special entrypoint (mentioned earlier), so that it's `engine.CreateDefaultValue(str)`, as `engine.Query(str)` is the only one that currently exists. The other alternative would be to serialize expressions, so that the integrator can store a byte array, which may be the easier option between the two. But there may be another option that I'm not thinking of.